### DISCUSSION TITLE
Is there a detailed explanation about the Preprocessors and models

### DISCUSSION THREAD

**Discussion by ykhasia** [lllyasviel/ControlNet#452] [Q&A] [answered]
There are many kinds of Preprocessors and models in ControlNet, but seems I can't find a good detailed explanation about the Preprocessors and models.
Such as the usage of the Preprocessors and models, What scenarios are these models used for?
The mapping relationship of the Preprocessors and models.
Is there a detailed explanation about the Preprocessors and models？ wiki or blog should be all right.
![image](https://github.com/lllyasviel/ControlNet/assets/52735731/d4749989-f3de-425d-8cad-efefcf2b1b9a)

**Comment by geroldmeisinger** [comment, marked as answer]
pretty much all of them are from external projects, thus you have to look there. you find additional info here [Adding Conditional Control to Text-to-Image Diffusion Models v1](https://arxiv.org/abs/2302.05543v1) - original ControlNet paper. HuggingFace also has a good overview of different ML models https://huggingface.co/docs/transformers/model_doc/blip-2

**Reply by ykhasia** [reply]
Got it. thank you
